src/strategies/cognito.bearer.strategy.js

```diff
@@ -0,0 +1,25 @@
+const Strategy = require('passport-http-bearer').Strategy;
+const cognitoAuthService = require('../utils/cognitoAuthService');
+const {UnauthorizedError} = require('../../utils/httpErrors');
+
+/**
+ * Bearer Strategy
+ *
+ * This strategy will handle requests with BearerTokens.  This is only a template and should be configured to
+ * your AuthZ server specifications.
+ *
+ * Requires ENV variables for introspecting the token
+ */
+module.exports.strategy = new Strategy(function (token, done) {
+    // https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html
+    cognitoAuthService.validate(token, function (err, result, decodedToken) {
+        if (err) {
+            return done(new UnauthorizedError(Error('Invalid token')));
```

Comment:
Should you return the error that actually occurred in `err` instead of sending a new invalid token? Otherwise you might have hard to track down bugs, especially if you aren't logging the actually error in the `validateToken` function.

Reply:
switched to using jwks-rsa